control/control_loop.go

```diff
@@ -295,6 +295,7 @@ func (l *Loop) Stop() {
 		close(l.ct.stop)
 		l.activeBackgroundWorkers.Wait()
 		l.running = false
+		l.cancel()
```

Comment:
the cancelctx is being used within the activeBackgroundWorkers, I think we need to cancel before waiting on those to complete if we believe the hanging is caused by this.

Reply:
I'll try switching it

Reply:
I don't see a difference in behavior but I don't think there's an issue with hanging anymore. But I want to make sure the loop context is closed correctly